File: Projects/TP06AdapterView/app/src/main/java/com/hyun/tp06adapterview/MainActivity.java
package com.hyun.tp06adapterview;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    ArrayList<Item> items = new ArrayList<Item>();

    MyAdapter adapter;
    RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

      items.add(new Item("ms01", "1",R.drawable.ms_01));
        items.add(new Item("ms02", "1",R.drawable.ms_02));
        items.add(new Item("ms03", "1",R.drawable.ms_03));
        items.add(new Item("ms04", "1",R.drawable.ms_04));
        items.add(new Item("ms05", "1",R.drawable.ms_05));
        items.add(new Item("ms06", "1",R.drawable.ms16));
        items.add(new Item("ms07", "1",R.drawable.ms_07));
        items.add(new Item("ms08", "1",R.drawable.ms_08));

        recyclerView= findViewById(R.id.recycler_view);
        adapter= new MyAdapter(items,this);
        recyclerView.setAdapter(adapter);




    }
}

File: Projects/Ex46KotlinRecyclerView/app/src/main/java/com/hyun/ex46kotlinrecyclerview/MainActivity.kt
package com.hyun.ex46kotlinrecyclerview

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {

    // 뷰 참조변수
    val recyclerView:RecyclerView by lazy { findViewById(R.id.recycler_view) }

    // 대량의 데이터를 저장하는 리스트
    var itemList: MutableList<Item> = mutableListOf()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //데이터를 추가하기.. -- 실무에서는 서버나. DB에서 값을 읽어옴.
        loadData()

        // 리사이클러뷰에 보여줄 아이템뷰를 만들어주는 아답터 객체 생성 및 설정
        recyclerView.adapter= MyAdapter(this, itemList)

        // 리사이클러뷰의 아이템 배치를 관리하는 관리자 설정
        recyclerView.layoutManager= LinearLayoutManager(this, LinearLayoutManager.VERTICAL,true)

    }//onCreate method

    // 대량의 데이터를 추가하는 메소드
    private fun loadData(){
        itemList.add(Item("sam","Hello.kotlin", R.drawable.bg_one01))
        itemList.add(Item("robin","Nice.kotlin", R.drawable.bg_one02))
        itemList.add(Item("kim","Good.kotlin", R.drawable.bg_one03))
        itemList.add(Item("man","Hello.jaca", R.drawable.bg_one04))
        itemList.add(Item("him","sad.java", R.drawable.bg_one05))
        itemList.add(Item("girl","Hello.kok", R.drawable.bg_one06))
    }


}//MainActivity class//